components/camel-vertx-http/src/main/java/org/apache/camel/component/vertx/http/VertxHttpComponent.java

```diff
@@ -0,0 +1,227 @@
+                        uriHttpUriAddress.getPort(),
+                        uriHttpUriAddress.getPath(),
+                        uriHttpUriAddress.getQuery(),
+                        uriHttpUriAddress.getFragment()),
+                        parameters);
+
+        configuration.setHttpUri(httpUri);
+
+        return endpoint;
+    }
+
+    @Override
+    public Producer createProducer(CamelContext camelContext, String host,
+            String verb, String basePath, String uriTemplate, String queryParameters, String consumes,
+            String produces, RestConfiguration configuration, Map<String, Object> parameters) throws Exception {
+        // avoid leading slash
+        basePath = FileUtil.stripLeadingSeparator(basePath);
+        uriTemplate = FileUtil.stripLeadingSeparator(uriTemplate);
+
+        // get the endpoint
+        String scheme = "vertx-http:";
+        String url = scheme + host;
+        if (!ObjectHelper.isEmpty(basePath)) {
+            url += "/" + basePath;
+        }
+        if (!ObjectHelper.isEmpty(uriTemplate)) {
+            url += "/" + uriTemplate;
+        }
+
+        RestConfiguration config = configuration;
+        if (config == null) {
+            config = CamelContextHelper.getRestConfiguration(getCamelContext(), null, scheme);
+        }
+
+        Map<String, Object> map = new HashMap<>();
+        // build query string, and append any endpoint configuration properties
+        if (config.getProducerComponent() == null || config.getProducerComponent().equals(scheme)) {
+            // setup endpoint options
+            if (config.getEndpointProperties() != null && !config.getEndpointProperties().isEmpty()) {
+                map.putAll(config.getEndpointProperties());
+            }
+        }
+
+        // get the endpoint
+        String query = URISupport.createQueryString(map);
+        if (!query.isEmpty()) {
+            url = url + "?" + query;
+        }
+
+        parameters = parameters != null ? new HashMap<>(parameters) : new HashMap<>();
+
+        // there are cases where we might end up here without component being created beforehand
+        // we need to abide by the component properties specified in the parameters when creating
+        // the component, one such case is when we switch from "http" to "https" component name
+        RestProducerFactoryHelper.setupComponentFor(url, camelContext, (Map<String, Object>) parameters.remove("component"));
+
+        VertxHttpEndpoint endpoint = camelContext.getEndpoint(url, VertxHttpEndpoint.class);
+        setProperties(endpoint, parameters);
+        String path = uriTemplate != null ? uriTemplate : basePath;
+        endpoint.getConfiguration().setHeaderFilterStrategy(new VertxHttpRestHeaderFilterStrategy(path, queryParameters));
+
+        // the endpoint must be started before creating the producer
+        ServiceHelper.startService(endpoint);
+
+        return endpoint.createProducer();
+    }
+
+    public Vertx getVertx() {
```

Comment:
I think in this case, we need to close/stop the vertx instance once the component stops